features/crc/soft/FastCRC.h

```diff
@@ -0,0 +1,202 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_FAST_CRC_H
+#define MBED_FAST_CRC_H
+
+#include "mbed_crc.h"
+#include "mbed.h"
+
+template <typename T>
+class FastCRC : public CRC
+{
+public:
+
+    /** Lifetime of the fast software CRC module
+     *
+     * Fast CRC module generates division table during initialization and uses table
+     * for runtime CRC generation from data input.
+     *
+     *  @param  polynomial  CRC polynomial
+     */
+    FastCRC(crc_polynomial_t polynomial) : _inital_value(0x0), _final_xor(0x0),
+                                           _reflect_data(false), _reflect_remainder(false) {
+        switch(polynomial) {
+            case CRC_7BIT_SD:
+                _polynomial = 0x9;             // x7+x3+1;
+                _width = 7;
+                break;
```

Comment:
Why not just provide polynomial/width/initial_value/reflect_data/reflect_remainder/final_xor all as constructor parameters with reasonable defaults?

If you want to get creative, you could do something like this:
``` cpp
// Presets for standard 32 bit CRC
#define CRC_32BIT 0x04C11DB7, 32, ~0, true, true, ~0

// Declare 32 bit CRC
SlowCRC<uint32_t> crc(CRC_32BIT);

// Declare non-standard 32 bit CRC
SlowCRC<uint32_t> crc(0x04C11DB7, 32, 0);
```

Reply:
Because not all of them are verified, I don't want to give impression that all CRC computations are implemented without verifying each of them. Ones which are used most are implemented and verified. Deviation from standard ones will give incorrect result.

Reply:
> Deviation from standard ones will give incorrect result.

They're all backed by the same algorithm though aren't they?

You could still say only the CRCs with matching defines are supported, and assert for invalid configurations in the constructor.